api/src/groups/test_groups.py

```diff
@@ -5,993 +5,816 @@
     # THEN we assume the correct status code
-    assert resp.status_code == 201
-    # THEN we assume the correct content of the group in the database
-    updated_group = auth_client.sqla.query(Group).filter_by(id=group.id).first()
-    for attr in ['name', 'description', 'manager_id', 'active']:
-        assert vars(updated_group)[attr] == payload[attr]
-
-    # THEN we assume the correct amount of members with the group in the database
-    queried_members = auth_client.sqla.query(Group).filter_by(id=group.id).first().members
-    queried_members = auth_client.sqla.query(Member).filter_by(group_id=group.id).filter_by(active=True).all()
-
-    print(f"members in database {queried_members}, members in the payload {payload['person_ids']}")
-
-    for queried_member in queried_members:
-        print(f"person id in database: {queried_member.person_id}")
-    assert len(queried_members) == len(payload['person_ids'])
-    # THEN we assume each member in the database match up with a member in the payload
-    for queried_member in queried_members:
-        assert queried_member.person_id in payload['person_ids']
-
-    # WHEN we update a non-existant group
-    resp = auth_client.patch(url_for('groups.update_group', group_id=-1), json=payload)
-    # THEN we expect an error code
-    assert resp.status_code == 404
+    assert resp.status_code == 204
 
-    # WHEN we update with an invalid manager
-    payload['manager_id'] = -1
-    resp = auth_client.patch(url_for('groups.update_group', group_id=group.id), json=payload)
+    # WHEN we delete a non-existent item
+    resp = auth_client.delete(url_for('groups.delete_group_type', group_type_id = -1))
     # THEN we expect an error
     assert resp.status_code == 404
 
+    # THEN we assume the number of group_types in the database to be the correct number
+    group_types = auth_client.sqla.query(GroupType).all()
+    assert len(group_types) == count - 1
 
-@pytest.mark.smoke
-def test_invalid_update_group(auth_client):
-    # GIVEN a database with a number of groups
-    count = random.randint(3, 11)
-    generate_managers(auth_client)
-    create_multiple_groups(auth_client.sqla, count)
-
-    original_group = auth_client.sqla.query(Group).first()
-    modified_group = group_object_factory(auth_client.sqla)
-
-    modified_group['invalid_field'] = 4
-
-    resp = auth_client.patch(url_for('groups.update_group', group_id=original_group.id), json=modified_group)
+# # ---- Group
 
-    # THEN we assume the incorrect status code
+@pytest.mark.smoke
+def test_create_group(auth_client):
+    # Test creating invalid group
+    # GIVEN invalid group to put in database
+    broken_group = {}
+    # WHEN database queried
+    resp = auth_client.post(
+        url_for('groups.create_group'), json=broken_group)
+    # THEN assert exception thrown
     assert resp.status_code == 422
-
+    # Test creating valid group
```

Comment:
In general, tests should test _just one thing_. Split this test into two: `test_create_invalid_group` and `test_create_valid_group`, for example.

Reply:
Got it! will fix it soon